cmd/provider/daemon.go

```diff
@@ -194,6 +196,38 @@ func daemonCommand(cctx *cli.Context) error {
 		defer bootstrapper.Close()
 	}
 
+	// setting up reframe server
+	var reframeSrv *reframeserver.Server
+	if len(cfg.Reframe.ListenMultiaddr) != 0 {
+		reframeAddr, err := cfg.Reframe.ListenNetAddr()
+		if err != nil {
+			return err
+		}
+
+		reframeSrv, err = reframeserver.New(
+			cfg.Reframe.CidTtl,
+			cfg.Reframe.ChunkSize,
+			cfg.Reframe.ProviderID,
+			cfg.Reframe.Addrs,
+			eng,
+			ds,
+			reframeserver.WithListenAddr(reframeAddr),
+			reframeserver.WithReadTimeout(cfg.Reframe.ReadTimeout),
+			reframeserver.WithWriteTimeout(cfg.Reframe.WriteTimeout),
+		)
+
+		if err != nil {
+			return err
+		}
+		log.Infow("reframe server initialized", "address", cfg.Reframe.ListenMultiaddr)
+
+		errChan := make(chan error, 1)
```

Comment:
this is a new errChan variable that shadows the other one and will not be read below

Reply:
great spot - removed new channel creation